### PR TITLE
Python: Add setup for contributing-python and tests for Semantic-Kernel-Python

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
Add setup tasks for contributing-python and tests for Semantic-Kernel-Python. The setup tasks install poetry and python packages. The tests include unit tests, integration tests, and all tests for Semantic-Kernel-Python.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -88,6 +88,57 @@
...
+      "command": "poetry",
+      "type": "shell",
+      "args": ["install"],
+      "options": {
+        "cwd": "${workspaceFolder}/python"
+      }
+    },
+    // Formatting
+    {
+      "label": "format (contributing-python-ruff)",
```
Comment (awharrison-28) [inline .vscode/tasks.json:123]: I suggest removing this one. `poetry run pre-commit...` already does ruff

**Review by awharrison-28 — CHANGES_REQUESTED** [review]

**Review by awharrison-28 — APPROVED** [review]

File: .vscode/tasks.json [hunk 2]
```diff
@@ -223,6 +264,58 @@
...
+      "group": "test",
+      "presentation": {
+        "reveal": "always",
+        "panel": "shared",
+        "group": "PR-Validate"
+      },
+      "options": {
+        "cwd": "${workspaceFolder}/python"
+      }
+    }
```
Comment (dmytrostruk) [inline .vscode/tasks.json:317]: VSCode shows me that comma is required here :)
image
```suggestion
    },
```

File: .vscode/tasks.json [hunk 3]
```diff
@@ -223,6 +264,58 @@
...
+      },
+      "options": {
+        "cwd": "${workspaceFolder}/python"
+      }
+    },
+    {
+      "label": "test (Semantic-Kernel-Python Integration)",
+      "command": "poetry",
+      "type": "shell",
+      "args": ["run", "pytest", "tests/integration","-k","${input:filter}",],
```
Comment (dmytrostruk) [inline .vscode/tasks.json:290]: I use Prettier, and on file save it automatically removes commas after last array item. GitHub also highlights this comma:
image

I'm wondering what formatter you use and should we setup some rules to use the same formatter for JSON files?

File: .vscode/tasks.json [hunk 4]
```diff
@@ -223,6 +264,58 @@
...
+      },
+      "options": {
+        "cwd": "${workspaceFolder}/python"
+      }
+    },
+    {
+      "label": "test (Semantic-Kernel-Python ALL)",
+      "command": "poetry",
+      "type": "shell",
+      "args": ["run", "pytest", "tests","-k","${input:filter}",],
```
Comment (dmytrostruk) [inline .vscode/tasks.json:306]: Same comma after last array item here, but not sure it's a problem at all.

File: .vscode/tasks.json (hunk 3 above)
Comment (lemillermicrosoft) [reply .vscode/tasks.json:290]: Yeah I use prettier too -- sometimes it's flaky for me though. Working now.

**Review by dmytrostruk — APPROVED** [review]

**Pull request merged**